lib/containers/urls.pm

```diff
@@ -51,64 +51,211 @@ sub get_opensuse_registry_prefix {
+            totest_registry   => sub {
+                "registry.opensuse.org/" . get_opensuse_registry_prefix . "opensuse/tumbleweed";
+            },
+            available_arch => ['x86_64', 'aarch64', 'ppc64le', 's390x', 'arm']
+        },
+        '15.1' => {
+            released_registry => sub { "registry.opensuse.org/opensuse/leap:15.1" },
+            totest_registry   => sub {
+                if (check_var('ARCH', 'x86_64')) {
+                    "registry.opensuse.org/opensuse/leap/15.1/images/totest/containers/opensuse/leap:15.1";
+                } elsif (check_var('ARCH', 'aarch64') || check_var('ARCH', 'arm')) {
+                    "registry.opensuse.org/opensuse/leap/15.1/arm/images/totest/containers/opensuse/leap:15.1";
+                } else {
+                    "";
+                }
+            },
+            available_arch => ['x86_64', 'aarch64', 'arm']
+        },
+        '15.2' => {
+            released_registry => sub { "registry.opensuse.org/opensuse/leap:15.2" },
+            totest_registry   => sub {
+                if (check_var('ARCH', 'x86_64')) {
+                    "registry.opensuse.org/opensuse/leap/15.2/images/totest/containers/opensuse/leap:15.2";
+                } elsif (check_var('ARCH', 'aarch64') || check_var('ARCH', 'arm')) {
+                    "registry.opensuse.org/opensuse/leap/15.2/arm/images/totest/containers/opensuse/leap:15.2";
+                } else {
+                    "";
+                }
+            },
+            available_arch => ['x86_64', 'aarch64', 'arm']
+        },
+        '15.3' => {
+            released_registry => sub { "registry.opensuse.org/opensuse/leap:15.3" },
+            totest_registry   => sub {
+                if (check_var('ARCH', 'x86_64')) {
+                    "registry.opensuse.org/opensuse/leap/15.3/images/totest/containers/opensuse/leap:15.3";
+                } elsif (check_var('ARCH', 'aarch64') || check_var('ARCH', 'arm')) {
+                    "registry.opensuse.org/opensuse/leap/15.3/arm/images/totest/containers/opensuse/leap:15.3";
+                } else {
+                    "";
+                }
+            },
+            available_arch => ['x86_64', 'aarch64', 'arm']
+        }
+    }
+);
+
+sub check_image_arch_in_registry {
+    my ($distri, $version, $arch) = @_;
+    (grep /\b$arch\b/, @{$registries{$distri}{$version}{available_arch}}) ? 1 : 0;
+}
+
 # Returns a tuple of image urls and their matching released "stable" counterpart.
 # If empty, no images available.
 sub get_suse_container_urls {
-    my $version    = shift // get_required_var('VERSION');
-    my $dotversion = $version =~ s/-SP/./r;                    # 15 -> 15, 15-SP1 -> 15.1
-    $dotversion = "${dotversion}.0" if $dotversion !~ /\./;    # 15 -> 15.0
-
+    my $version         = shift // get_required_var('VERSION');
+    my $arch            = shift // get_required_var('ARCH');
+    my $distri          = shift // get_required_var('DISTRI');
```

Comment:
I would prefer named parameters like:
```suggestion
    my (%args) = @_;
    $args{version} //= get_required_var('VERSION');
    $args{arch} //=  get_required_var('ARCH');
    $args{distri} //=  get_required_var('DISTRI');
```

Reply:
This will add a few more files in the PR FYI

Reply:
*this change is ready for review* but i will have to make another commit for the affected files. leave the conversation open

Reply:
*additional commit added* 
thanks @cfconrad you can close this conversation if you are ok with the latest commit. i touched only files with a positional parameter.